### PR TITLE
Initial pull-request

### PR DESCRIPTION
**Pull request by georgiybykov**
- Create new application
- Add and configure linter
- Check test coverage of the code
- Type checking
- Add GitHub Actions configuration

### CODE REVIEW

File: lib/hw_app.ex [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def hello(str), do: "Hello #{str}!"
+
+  @doc """
+  Greeting to the World
+
+    iex> HWApp.hello()
+    "Hello World!"
+  """
+  @spec hello() :: String.t()
+  def hello, do: "Hello World!"
```
Comment (am-kantox) [inline lib/hw_app.ex]: We can have default parameters in _Elixir_:

```elixir
@spec hello(String.t()) :: String.t()
def hello(str \\ "World"), do: "Hello #{str}!"
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+ - configured GitHub Actions.
+ 
+##### Dependecies:
+
+```elixir
+  defp deps do
+    [
+      {:dialyxir, "~> 1.0", only: [:dev, :test], runtime: false},
+      {:credo, "~> 1.6", only: [:dev, :test], runtime: false},
+      {:credo_contrib, "~> 0.2.0", only: [:dev, :test], runtime: false},
```
Comment (am-kantox) [inline README.md]: Literally all known to me _Elixir_ libraries use semantic versioning, hence
```suggestion
      {:credo_contrib, "~> 0.2", only: [:dev, :test], runtime: false},
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+# HWApp [![CI Tests](https://github.com/georgiybykov/hw_app/actions/workflows/elixir.yml/badge.svg)](https://github.com/georgiybykov/hw_app/actions)
 
-Hello World application as tutrial one with test and CI.
+Hello World application as tutrial one with:
+ - linter;
+ - type checking;
+ - tests;
+ - configured GitHub Actions.
+ 
+##### Dependecies:
```
Comment (am-kantox) [inline README.md]: I doubt this is actually needed to be exposed in `README` :)

File: mix.exs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    [
+      extra_applications: [:logger]
+    ]
+  end
+
+  defp deps do
+    [
+      {:dialyxir, "~> 1.0", only: [:dev, :test], runtime: false},
+      {:credo, "~> 1.6", only: [:dev, :test], runtime: false},
+      {:credo_contrib, "~> 0.2.0", only: [:dev, :test], runtime: false},
```
Comment (am-kantox) [inline mix.exs]: Literally all known to me _Elixir_ libraries use semantic versioning, hence
```suggestion
      {:credo_contrib, "~> 0.2", only: [:dev, :test], runtime: false},
```

File: test/hw_app_test.exs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  use ExUnit.Case
+  doctest HWApp
+
+  describe "#hello" do
+    test "when the name of somebody has been passed" do
+      assert HWApp.hello('Name') == "Hello Name!"
+    end
+
+    test "when the name has not been provided" do
+      assert HWApp.hello() == "Hello World!"
```
Comment (am-kantox) [inline test/hw_app_test.exs]: This is already tested with a respective doctest, no need to duplicate the test here.

**Review by am-kantox — APPROVED** [review]

File: lib/hw_app.ex (hunk 1 above)
Comment (georgiybykov) [reply lib/hw_app.ex]: Thank you!

File: README.md (hunk 3 above)
Comment (georgiybykov) [reply README.md]: Of course! :D

File: test/hw_app_test.exs (hunk 5 above)
Comment (georgiybykov) [reply test/hw_app_test.exs]: Okay, thanks!

File: mix.exs (hunk 4 above)
Comment (georgiybykov) [reply mix.exs]: I think both examples use semantic versioning, however I haven't noticed that deps consist of major and minor versions without patch one. This is true. Thanks!

**Pull request merged**